src/lib.rs

```diff
@@ -0,0 +1,564 @@
+use std::sync::{
+    atomic::{AtomicBool, Ordering},
+    Arc,
+};
+
+use concordium_rust_sdk::{
+    postgres::{self, DatabaseClient},
+    types::{hashes::BlockHash, AbsoluteBlockHeight},
+    v2::{self, FinalizedBlockInfo},
+};
+use structopt::StructOpt;
+use thiserror::Error;
+use tonic::{async_trait, transport::ClientTlsConfig};
+
+const MAX_CONNECT_ATTEMPTS: u64 = 6;
+
+/// A collection of variables supplied to [`run_service`]. These determine how
+/// the service runs with regards to connections to concordium node(s), db, and
+/// logging.
+#[derive(StructOpt)]
+pub struct SharedIndexerArgs {
```

Comment:
This, the actual command-line arguments don't really belong to a library. The name of environment variables is not

I was going to suggest what other things we can do but ran out of time at the moment. Let's discuss it offline perhaps.

Reply:
Yes, I agree. Will adapt